### PR TITLE
Feature/auth_jwt

### PR DESCRIPTION
**Pull request by vibhu0100**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+class ApplicationController < ActionController::API
+    before_action :authorized
+
+    def authorized
+        render json: {message: "Please log in"},status: :unauthorized unless logged_in?
+    end
+    def logged_in?
+    !!logged_in_user
+    end
+
```
Comment (nehanakrani-simformsolutions) [inline app/controllers/application_controller.rb]: Check indentation

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            token=auth_header.split(' ')[1]
+            begin
+                JWT.decode(token, 'secret',true,algorith: 'HS256')
+
+            rescue JWT::DecodeError
+                nil
+
+                
+            end
+        end
```
Comment (nehanakrani-simformsolutions) [inline app/controllers/application_controller.rb]: Check indentation

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        @user=User.find_by(id: user_id)
+    end
+end
+
+
+    def decodeds_token
+        if auth_header
+            token=auth_header.split(' ')[1]
+            begin
+                JWT.decode(token, 'secret',true,algorith: 'HS256')
```
Comment (nehanakrani-simformsolutions) [inline app/controllers/application_controller.rb]: `token, secret, true, algorith: 'HS256'` make sure after coma space will there in everywhere

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        end
+
+    end
+
+    def encode_token(payload)
+    JWT.encode(payload, 'secret')
+  end
+
+  def auth_header
+    # { Authorization: 'Bearer <token>' }
```
Comment (nehanakrani-simformsolutions) [inline app/controllers/application_controller.rb]: make sure to remove unused code

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    request.headers['Authorization']
+  end
+
```
Comment (nehanakrani-simformsolutions) [inline app/controllers/application_controller.rb:43]: Why you add extra spaces remove it

File: app/controllers/application_controller.rb [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+
```
Comment (nehanakrani-simformsolutions) [inline app/controllers/application_controller.rb]: Why you add extra spaces

File: app/controllers/application_controller.rb [hunk 7]
```diff
@@ -0,0 +1,46 @@
+class ApplicationController < ActionController::API
+    before_action :authorized
+
+    def authorized
+        render json: {message: "Please log in"},status: :unauthorized unless logged_in?
```
Comment (nehanakrani-simformsolutions) [inline app/controllers/application_controller.rb]: comma after space missing

File: app/controllers/application_controller.rb [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+                
+            end
+        end
+
+    end
+
+    def encode_token(payload)
+    JWT.encode(payload, 'secret')
+  end
```
Comment (nehanakrani-simformsolutions) [inline app/controllers/application_controller.rb:27]: Check indentation

File: app/controllers/users_controller.rb [hunk 9]
```diff
@@ -0,0 +1,47 @@
+class UsersController < ApplicationController
+    before_action :authorized, only: [:auto_login]
+
+    def create
+        @user=User.new(user_params)
+        if @user.save
+            token=encode_token({user_id: @user.id})
```
Comment (nehanakrani-simformsolutions) [inline app/controllers/users_controller.rb:7]: missing space

Comment (nehanakrani-simformsolutions) [inline app/controllers/users_controller.rb]: missing space

File: app/controllers/users_controller.rb [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+    def authenticate
+        @user=User.find_by(email: params[:email])
+
+        if @user.password.eql?(params[:password])
+            token=encode_token({user_id: @user.id})
+            render json: {user: @user, token: token}
+        else
+            render json: {error: "Invalid"}
+        end
+=begin
```
Comment (nehanakrani-simformsolutions) [inline app/controllers/users_controller.rb:24]: Remove it if not using

File: app/controllers/users_controller.rb [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+        if user
+            if user.password==params[:password]
+                render json: "Authenticted"
+            else
+                render json: "Enter correct password"
+            end
+        else
+            render json: "You are not registered"
+        end
+=end
```
Comment (nehanakrani-simformsolutions) [inline app/controllers/users_controller.rb]: Remove it if not using make sure code will be clean

File: app/controllers/users_controller.rb [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+            render json: {error: "Enter Proper data"}
+        end
+
+
+    end
+    def authenticate
+        @user=User.find_by(email: params[:email])
+
+        if @user.password.eql?(params[:password])
+            token=encode_token({user_id: @user.id})
```
Comment (nehanakrani-simformsolutions) [inline app/controllers/users_controller.rb]: Missing space

File: bin/bundle [hunk 13]
```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+#
+# This file was generated by Bundler.
```
Comment (nehanakrani-simformsolutions) [inline bin/bundle]: What is use of this file, why you pushed it

File: bin/bundle.cmd [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+# This file was generated by Bundler.
+#
+# The application 'bundle' is installed as part of a gem, and
+# this file is here to facilitate running it.
+#
+
+require "rubygems"
+
+m = Module.new do
+  module_function
```
Comment (nehanakrani-simformsolutions) [inline bin/bundle.cmd]: same here, What is used if this file, why you pushed it

Comment (nehanakrani-simformsolutions) [reply bin/bundle.cmd]: same as all bin/ folder files any reason to push or you're missing adding git ignore file
NOTE: understand use of `gitignore`

File: log/development.log [hunk 15]
```diff
@@ -0,0 +1,184 @@
...
+  [1m[35m (0.9ms)[0m  [1m[34mSELECT "schema_migrations"."version" FROM "schema_migrations" ORDER BY "schema_migrations"."version" ASC[0m
+Migrating to CreateUsers (20210303064413)
+  [1m[35m (0.8ms)[0m  [1m[34mSELECT sqlite_version(*)[0m
+  [1m[35m (0.3ms)[0m  [1m[34mSELECT "schema_migrations"."version" FROM "schema_migrations" ORDER BY "schema_migrations"."version" ASC[0m
+Migrating to CreateUsers (20210303064413)
+  [1m[36mTRANSACTION (0.2ms)[0m  [1m[36mbegin transaction[0m
+  [1m[35m (2.1ms)[0m  [1m[35mCREATE TABLE "users" ("id" integer PRIMARY KEY AUTOINCREMENT NOT NULL, "name" varchar, "email" varchar, "password" varchar, "created_at" datetime(6) NOT NULL, "updated_at" datetime(6) NOT NULL)[0m
+  [1m[36mActiveRecord::SchemaMigration Create (1.1ms)[0m  [1m[32mINSERT INTO "schema_migrations" ("version") VALUES (?)[0m  [["version", "20210303064413"]]
+  [1m[36mTRANSACTION (196.9ms)[0m  [1m[36mcommit transaction[0m
+  [1m[36mActiveRecord::InternalMetadata Load (0.7ms)[0m  [1m[34mSELECT "ar_internal_metadata".* FROM "ar_internal_metadata" WHERE "ar_internal_metadata"."key" = ? LIMIT ?[0m  [["key", "environment"], ["LIMIT", 1]]
```
Comment (nehanakrani-simformsolutions) [inline log/development.log]: Never push your log file in GitHub

Use cmd line to push code not drag and drop to add files in such case unwanted files may push
Make sure you add gitignore and remove all unwanted files.

**Comment by nehanakrani-simformsolutions** [comment]
- Your commit name, title and desc should  be informatic
- code indentation must be proper and used in two space conversation default for .rb file
- Definition name must be clear
- Must be add git ignore file in project

**Review by nehanakrani-simformsolutions — COMMENTED** [review]
> Must to follow:
> 
> * Your commit name, title and desc should  be informatic
> * code indentation must be proper and used in two space conversation default for .rb file
> * Definition name must be clear
> * Must be add git ignore file in project

File: log/development.log (hunk 15 above)
Comment (vibhu0100) [reply log/development.log]: okay

**Pull request merged**